File: gaia-sdk-python/gaia_sdk/graphql/request/type/BehaviourStatesPerDay.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class BehaviourStatesPerDay(list):

    def date(self):
        self.append(lambda x: "date")

    def running(self):
        self.append(lambda x: "running")

    def success(self):
        self.append(lambda x: "success")

    def waiting(self):
        self.append(lambda x: "waiting")

    def failed(self):
        self.append(lambda x: "failed")

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))

File: gaia-sdk-javascript/src/graphql/request/type/BehaviourStatesPerDay.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

export class BehaviourStatesPerDay extends Array<(_:VariableRegistry) => string> {
public _typeName = "BehaviourStatesPerDay";
    public date = () => { 
        this.push(_ => "date")
    };

    public running = () => { 
        this.push(_ => "running")
    };

    public success = () => { 
        this.push(_ => "success")
    };

    public waiting = () => { 
        this.push(_ => "waiting")
    };

    public failed = () => { 
        this.push(_ => "failed")
    };

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

